xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -1021,7 +1147,8 @@ private void handleEdsResponse(DiscoveryResponse edsResponse) {
     public void run() {
       startRpcStream();
       if (configWatcher != null) {
-        adsStream.sendXdsRequest(ADS_TYPE_URL_LDS, ImmutableList.of(ldsResourceName));
+        adsStream.sendXdsRequest(ADS_TYPE_URL_LDS, ldsResourceName != null
```

Comment:
Why change this? When `configWatcher` is not null, `ldsResourceName` will never be null.

Reply:
You are right. I should have a separate if clause with `listenerWatcher != null` and use `ListenerResourceFetchTimeoutTask`. Will make that change and see if a test is needed